libqtile/bar.py

```diff
@@ -175,34 +177,50 @@ def __init__(self, widgets, size, **config):
         self.size_calculated = 0
         self._configured = False
 
+        if isinstance(self.margin, int):
+            self.margin = [self.margin] * 4
+
+        if isinstance(self.border, int):
+            self.border = [self.border] * 4
+
         self.queued_draws = 0
 
     def _configure(self, qtile, screen):
         Gap._configure(self, qtile, screen)
 
-        if self.margin:
-            if isinstance(self.margin, int):
-                self.margin = [self.margin] * 4
+        if sum(self.margin) or sum(self.border):
+
+            if isinstance(self.border_colour, str):
+                self.border_colour = [self.border_colour] * 4
```

Comment:
Perhaps this should accept other forms i.e. `Union[str, Tuple[int, int, int], Tuple[int, int, int, float]]`